### DISCUSSION TITLE
ContentItem Conversion

### DISCUSSION THREAD

**Discussion by brogits** [OrchardCMS/OrchardCore#15845] [Q&A]
Is it Possible to convert ContentItem to a given Type?
I'm trying to use this code

`var model = memberContentItem.As<MembersEntryViewModel>()`

where memberContentItem is of type ContentItem.
MembersEntryViewModel is same model that was use to save the content item using the following code:

```
member.Apply("FirstName", new TextField() { Text = model.FirstName });
member.Apply("LastName", new TextField() { Text = model.LastName }); 
ember.Apply("PhoneNumber", new TextField() { Text = model.PhoneNumber }); 
member.Apply("Email", new TextField() { Text = model.Email });

await _contentManager.CreateAsync(member);
```

There is no error but I'm getting a null result.

**Comment by sebastienros** [comment]
If the properties of the content item map to the ones of the view model yes. But it looks like it's not the case.
I assume this would work if you did `member.Data.Firstname = model.Firstname`.

If you look at the json representation of your content item it will be clear why.

**Comment by brogits** [comment]
I did try to match my model with the content Item and still I'm getting a null result with 

`var model = contentItem.As<Model.MemberModel>();`

here is the contentItem
```
{{
  "FirstName": {
    "Text": "Ricky"
  },
  "LastName": {},
  "PhoneNumber": {
    "Text": "123-0987"
  },
  "Email": {},
  "Status": {
    "Text": "Pending"
  },
  "Employer": {
    "Text": "N/A"
  },
  "SupportWorker": {},
  "Created": {
    "Value": "2019-05-23T03:58:29.1573024Z"
  },
  "WorkHistories": {
    "ContentItems": [
      {
        "ContentItemId": "4h2x4684c99xp2xtrfvngxqswb",
        "ContentItemVersionId": null,
        "ContentType": "WorkHistory",
        "DisplayText": null,
        "Latest": false,
        "Published": false,
        "ModifiedUtc": null,
        "PublishedUtc": null,
        "CreatedUtc": null,
        "Owner": null,
        "Author": null,
        "EmployerName": {
          "Text": "tester"
        },
        "Position": {
          "Text": "upright"
        },
        "ToPresent": {},
        "FromDate": {
          "Value": "2019-05-23T00:00:00Z"
        },
        "ToDate": {
          "Value": "2019-05-23T00:00:00Z"
        }
      }
    ]
  }
}}
```
my model
```
public class MemberModel : ContentPart
    {
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string Email { get; set; }
        public string Status { get; set; }
        public string Employer { get; set; }
        public string SupportWorker { get; set; }
        public DateTime Created { get; set; }

        public List<WorkHistory> WorkHistories{ get; set; }
    }
```

**Comment by brogits** [comment]
@sebastienros : my goal hear is to Map the ContentItem to my MemberModel. Is there a proper way to do it in OC?

**Comment by sebastienros** [comment]
Most of it should be doable without any code at all, just by using existing fields and parts.

If you want to create a custom part, then you you shouldn't use any `TextField` as you will provide the editor template and driver. 

You can mix a custom editor and other fields though, in that case your Migration will add some fields to your part.

But my suggestion would be to only use dynamic fields and parts in a custom content type.
